core/src/toga/documents.py

```diff
@@ -16,7 +16,7 @@ def __init__(
         self,
         path: str | Path,
         document_type: str,
-        app: App = None,
+        app: App | None = None,
     ):
         """Create a new Document.
```

Comment:
Can `app` really be `None` here?

The implementation is missing several checks for `None` if that's the case.

Reply:
I think this is an oversight. I'm pretty sure `app` will always be specified and non-None.